API/ev3_timer.private.h

```diff
@@ -0,0 +1,323 @@
+/** \file ev3_timer.private.h
+ * \brief Private declarations of the ev3_timer module
+ *
+ * This file contains implementation-internal declarations
+ * that are not safe to use by user programs. For that purpose,
+ * please see the ev3_timer.h header.
+ *
+ * License:
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/
+ *
+ * The Initial Developer of this code is Jakub Vanek.
+ * Portions created by Jakub Vanek are Copyright (C) 2020 Jakub Vanek.
+ * All Rights Reserved.
+ *
+ * ----------------------------------------------------------------------------
+ *
+ * \author Jakub Vanek (linuxtardis_at_gmail.com)
+ * \date 2020-02-11
+ * \version 1
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifndef ev3_timer_private_h
+#define ev3_timer_private_h
+
+
+////////////////////////////////////////////////////////////////
+// INCLUDES
+
+#include "ev3_timer.h"
+#include <unistd.h>
+#include <errno.h>
+#include <stdio.h>
+#include <pthread.h>
+#include <sys/time.h>
+#include <sys/timerfd.h>
+#include <sys/eventfd.h>
+#include <sys/epoll.h>
+
+////////////////////////////////////////////////////////////////
+// MACROS
+
+//! initial capacity of each callback queue
+#define INITIAL_CALLBACKS  4
+#define TERMINATOR UINT32_MAX
+
+//! Log error using perror()
+#define REPORT(info) perror(__FILE__ ": " info)
+
+//! Log error using perror() & return false
+#define REPORT_AND_RETURN(info, value) \
+    do {                \
+        REPORT(info);   \
+        return value; \
+    } while(0)
+
+//! Log error using perror() & return false on value -1
+#define RETURN_ON_FAIL(x, info, value)    \
+    do {                      \
+        int _value = (x);     \
+        if (_value < 0) {     \
+            REPORT_AND_RETURN(info, value); \
+        }                     \
+    } while(0)
+
+//! Log error using perror() & set a variable to a given value if x == -1
+#define SET_ON_FAIL(x, info, var, value) \
```

Comment:
This is hardly more readable. Please use normal flow control.

Reply:
Oki, I will remove this macro.

Reply:
Resolved in 620130dbedef6e00dd6d2c55d062508f6af9967a :+1: